### PR TITLE
Add template argument signature help

### LINKED ISSUE #1348: Show alias signature/docs in IDE
Currently, we resolve to Type before computing documentation so we only show the docs on the target of an alias, not the ones on the alias itself

### LINKED ISSUE #1789: [Feature] Context or hover-help for template parameters
It would be helpful to get context or hover-help for template parameters when instantiating a template. Currently there is no help shown when the opening "<" is entered following a template name.

image

### PR DESCRIPTION
**Pull request by nguerrera**
Also show alias doc comments in IDE

Fix #1789
Fix #1348

### CODE REVIEW

**Comment by nguerrera** [comment]
Diff is easier to read without whitespace sensitivity.

File: packages/compiler/server/completion.ts
Comment (timotheeguerin) [inline packages/compiler/server/completion.ts:1]: Seems to have issue if you press space after `,` when trying to complete the 2nd parameter https://cadlplayground.z22.web.core.windows.net/prs/1983/?c=LyoqCiAqIFRoaXMgaXMgZm9vxA9AdGVtcGxhdGUgQSB0xxtBzhlCyRlCCiAqLwphbGlhcyBGb288QSwgQj4gPSBBICYgQjsKCgrHHFQxID3FIXt9LA%3D%3D

Comment (nguerrera) [reply packages/compiler/server/completion.ts:1]: Looking. Thanks for trying this.

Comment (timotheeguerin) [reply packages/compiler/server/completion.ts:1]: ah yeah I have the same thing happening, this make sense, lets file another issue then.

File: packages/compiler/server/type-signature.ts [hunk 1]
```diff
@@ -36,7 +51,7 @@ export function getTypeSignature(type: Type): string {
     case "Number":
       return `(number)\n${fence(type.value.toString())}`;
     case "Intrinsic":
-      return `(intrinsic)\n${fence(type.name)}`;
+      return "";
```
[the remark this answers is not in the document]
Comment (timotheeguerin) [reply packages/compiler/server/type-signature.ts:54]: was just thinking when null is used inside a property type or operation response it should show up on the signature no?

but this seems to still be working so just maybe doesn't call back to this recursively

**Review by timotheeguerin — APPROVED** [review]

File: packages/compiler/server/completion.ts
Comment (nguerrera) [reply packages/compiler/server/completion.ts:1]: Filed #1990

**Pull request merged**